lib/smack/IntegerOverflowChecker.cpp

```diff
@@ -95,7 +95,12 @@ Value* IntegerOverflowChecker::createResult(Value* v, int bits, Instruction* i)
  * overflow occured as indicated by flag.
  */
 void IntegerOverflowChecker::addCheck(Function* co, Value* flag, Instruction* i) {
+  /*
+  TODO: something is very weird here: if we enable the following `arg` definition,
+        there will be a segmentation fault with O2
```

Comment:
This is the part we don't understand. We thought the old implementation should be fine but somehow observed segmentation faults. We leave this comment so that maybe one day we can figure this out.

Reply:
Are they semantically equivalent?
I doubt we'll have time to look into this in the future, and so let's remove this comment.

Reply:
> Are they semantically equivalent?
> I doubt we'll have time to look into this in the future, and so let's remove this comment.

We thought so. Ok, let me remove it.

Reply:
Done.